File: by-python/py-boj/1912-연속합.py
def largest_sum(arr):
    dp = [0] * n
    dp[0] = arr[0]

    for i in range(1, n):
        dp[i] = max(dp[i - 1] + arr[i], arr[i])
    return max(dp)


if __name__ == '__main__':
    n = int(input())
    arr = list(map(int, input().split()))

    print(largest_sum(arr))

File: by-kotlin/kt-boj/solutions/1912-연속합.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import kotlin.math.max

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val n = readLine().toInt()
    val arr = readLine().split(" ").map { it.toInt() }

    print(largestSum(arr, n))
}

fun largestSum(arr: List<Int>, n: Int): Int {
    val dp = Array(n) { 0 }
    dp[0] = arr[0]

    for (i in 1 until n)
        dp[i] = max(dp[i - 1] + arr[i], arr[i])
    return dp.maxOrNull()!!
}
